src/common/pico_time/time.c

```diff
@@ -259,13 +259,13 @@ static void alarm_pool_irq_handler(void) {
                 index = next;
             }
         }
-        now = (int64_t) ta_time_us_64(timer);
         earliest_index = pool->ordered_head;
         if (earliest_index < 0) break;
         // need to wait
         alarm_pool_entry_t *earliest_entry = &pool->entries[earliest_index];
         earliest_target = earliest_entry->target;
         ta_set_timeout(timer, timer_alarm_num, earliest_target);
+        now = (int64_t) ta_time_us_64(timer);
```

Comment:
Maybe get rid of "now" altogether and just call ta_time_us_64 in the two places where its needed?

Reply:
sure; it avoids now getting stale ;-)